lib/collection/proxy-config.js

```diff
@@ -0,0 +1,114 @@
+var _ = require('../util').lodash,
+    Property = require('./property').Property,
+    Url = require('./url').Url,
+
+    E = '',
+    MATCH_ALL = '*',
+
+    ProxyConfig;
+
+
+/**
+* The following is the object structure accepted as constructor parameter while calling `new ProxyConfig(...)`. It is
+* also the structure exported when {@link Property#toJSON} or {@link Property#toObjectResolved} is called on a
+* Proxy instance.
+* @typedef ProxyConfig~definition
+*
+* @property {String=} [url = *] The match for which the proxy needs to be configured. `*` represents select all.
+* @property {Url=} [server = new Url()] The proxy server url.
+* @property {Boolean=} [tunnel = false] The tunneling option for the proxy request.
+*
+* @example <caption>JSON definition of an example proxy object</caption>
+* {
+*     match: 'https://example.com',
+*     server: 'https://proxy.com',
+*     tunnel: true
```

Comment:
we should add (for sake of example) `disabled: true // to bypass usage of this proxy`